pkg/controller/master/upgradelog/common.go

```diff
@@ -460,6 +480,29 @@ func setUpgradeLogArchiveReady(upgradeLog *harvesterv1.UpgradeLog, archiveName s
 	return fmt.Errorf("archive %s of %s not found", archiveName, upgradeLog.Name)
 }
 
+func setLoggingOperatorSource(upgradeLog *harvesterv1.UpgradeLog, source string) {
+	if upgradeLog.Annotations == nil {
+		upgradeLog.Annotations = make(map[string]string, 1)
+	}
+	upgradeLog.Annotations[util.UpgradeLogLoggingOperatorSource] = source
+}
+
+func getLoggingImageSourceHelmChart(upgradeLog *harvesterv1.UpgradeLog) (string, string, error) {
+	src := upgradeLog.Annotations[util.UpgradeLogLoggingOperatorSource]
+	if src == "" {
+		return "", "", fmt.Errorf("faild to get annotation %v from upgradeLog %v/%v", util.UpgradeLogLoggingOperatorSource, upgradeLog.Namespace, upgradeLog.Name)
+	}
+	if src == util.RancherLoggingName {
+		return util.CattleLoggingSystemNamespaceName, util.RancherLoggingName, nil
+	}
+	// format: hvst-upgrade-l5875-upgradelog-operator
+	if src == name.SafeConcatName(upgradeLog.Name, util.UpgradeLogOperatorComponent) {
+		return util.CattleLoggingSystemNamespaceName, src, nil
+	}
```

Comment:
Is there a reason we separate them into two conditions? It seems the return stuff could be the same.

Reply:
The `rancher-logging` is a fixed name, and upgradeLog operator is a dyname name like `hvst-upgrade-l5875-upgradelog-operator`, only them two are valid; and the returns value happen to use the same namespace, but may have different namespace

separate them is to let this to be even noticable;  merge them into  one if is also ok.

```
if src == util.RancherLoggingName || name.SafeConcatName(upgradeLog.Name, util.UpgradeLogOperatorComponent) {
```